src/bootstrap/doc.rs

```diff
@@ -620,19 +625,86 @@ impl Step for Rustc {
         let mut cargo = builder.cargo(compiler, Mode::Librustc, target, "doc");
         compile::rustc_cargo(build, &mut cargo);
 
-        if build.config.compiler_docs {
```

Comment:
Could you expand a bit on how come this step was separated? I'd naively expect it to be ok if it continued to just be one step that internally dispatched to all crates or just proc_macro

Reply:
When the standard documentation is being generated - this is done by merging the standard library, test and procedural macro documentation (normally, the `compiler_docs` flag would add everything else here too). This was achieved through symlinking `docs` to a shared folder for those three steps.

In order to produce the compiler documentation such that we would still have standard documentation separately which contains the same crates as it did pre-PR, I had to ensure that the symlink directory was different when generating all the compiler documentation. This is because the distribution step would a copy from that generated documentation and that would contain the compiler documentation too if it wasn't generated into somewhere else (this is because all documentation steps happen before dist steps).

(It occurs to me now when writing this that since each doc step copies from the shared symlink directory to its output directory at the end of each step that we could still use the same shared folder which might speed things up. This is because the dist step for standard documentation will use the folder that was copied before compiler documentation was added - will try this).

I assumed that we'd want the standard documentation to still include the `proc_macro` crate and so I kept one step that would generate that into the shared location and created another that would generate the compiler documentation into the separate location. These two locations then produce the two tarballs in the dist step.

Reply:
Updated that commit to contain the improvement I noticed in the above comment.